model/labels/labels_stringlabels.go

```diff
@@ -23,19 +23,105 @@ import (
 	"github.com/cespare/xxhash/v2"
 )
 
+// List of labels that should be mapped to a single byte value.
+// Obviously can't have more than 256 here.
+var mappedLabels = [256]string{
```

Comment:
Suggest to keep this under 128 so that the index can be varint-encoded in a single byte.
That probably gives a speed boost on decode, since it doesn't have to loop on the varint.

Reply:
Do you mean using the first (size) byte values 128-255?
https://github.com/prometheus/prometheus/blob/main/model/labels/labels_stringlabels.go#L34-L39

I did use a separate byte as I wasn't entirely sure why the shortest variant is for 0-127 lengths, rather than 0-255, and whether it's safe to use the rest of the range in the size byte.
If it is then I can refactor the code to use mapping for size byte values 128-255 instead of storing the index in a separate byte.

Reply:
Sorry I misread the code.  Your encoding is fine, it always uses 1 byte for the mapped labels.

I didn't expect the code to call `encodeSize` in the case of 0.

Varint always uses the 7 bits; just if the top bit it set it reads the next byte too.  I don't think it's worth getting cleverer than this.